app/src/main/java/com/example/mystore/ApiHelper.kt

```diff
@@ -1,7 +1,8 @@
 package com.example.mystore
 
-import com.example.mystore.models.ProductModel
+import com.example.mystore.model.ProductModel
+import retrofit2.Response
 
 interface ApiHelper {
-    suspend fun getProduct(): ProductModel
+    suspend fun getProduct( productId: Int?, populate: String? ) : Response<ProductModel>?
```

Comment:
Retrofit always returns a response, it might be an unsuccessful, but it will be a response, hence no need for the `?` at the end. This way you won't need the cast in the viewmodel later on, too.